versions.sh

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env bash
+set -Eeuo pipefail
+
+# we will support at most two entries in each of these lists, and both should be in descending order
+supportedDebianSuites=(
+	bookworm
+)
+supportedAlpineVersions=(
+	3.18
+)
+defaultDebianSuite="${supportedDebianSuites[0]}"
+declare -A debianSuites=(
+	[11]='' # https://github.com/docker-library/postgres/issues/582 😬
+)
+defaultAlpineVersion="${supportedAlpineVersions[0]}"
+declare -A alpineVersions=(
+	#[14]='3.16'
+)
+
+gosuVersion='1.16'
+
+cd "$(dirname "$(readlink -f "$BASH_SOURCE")")"
+
+versions=( "$@" )
+if [ ${#versions[@]} -eq 0 ]; then
+	versions=( */ )
+	json='{}'
+else
+	json="$(< versions.json)"
+fi
+versions=( "${versions[@]%/}" )
+
+packagesBase='https://raw.githubusercontent.com/redis/redis-hashes/master/README'
+
+declare -A packages=
+_raw_package_list() {
+	curl -fsSL "$packagesBase"
+}
+fetch_package_list() {
+	local -; set +x # make sure running with "set -x" doesn't spam the terminal with the raw package lists
+
+	# normal (GA) releases end up in the "main" component of upstream's repository
+	if [ -z "${packages}" ]; then
+		packages="$(_raw_package_list)"
+	fi
+}
+get_version() {
+	local version="$1"; shift
+	versionInfo=$(grep "redis-${version}" <<< "${packages}" | tail -1)
+
+	rcVersion="${version%-rc}"
+
+	line="$(
+		awk '
+			{ gsub(/^redis-|[.]tar[.]gz$/, "", $2) }
+			$1 == "hash" && $2 ~ /^'"$rcVersion"'([.]|$)/ { print }
+		' <<< "$versionInfo" \
+			| sort -rV \
+			| head -1
+	)"
+
+	if [ -n "$line" ]; then
+		fullVersion="$(cut -d' ' -f2 <<<"$line")"
+		downloadUrl="$(cut -d' ' -f5 <<<"$line")"
+		shaHash="$(cut -d' ' -f4 <<<"$line")"
+		shaType="$(cut -d' ' -f3 <<<"$line")"
+	elif [ "$version" != "$rcVersion" ] && fullVersion="$(
+			git ls-remote --tags https://github.com/redis/redis.git "refs/tags/$rcVersion*" \
+				| cut -d/ -f3 \
+				| cut -d^ -f1 \
+				| sort -urV \
+				| head -1
+	)" && [ -n "$fullVersion" ]; then
+		downloadUrl="https://github.com/redis/redis/archive/$fullVersion.tar.gz"
+		shaType='sha256'
+		shaHash="$(curl -fsSL "$downloadUrl" | "${shaType}sum" | cut -d' ' -f1)"
+	else
+		echo >&2 "error: full version for $version cannot be determined"
+		exit 1
+	fi
+	[ "$shaType" = 'sha256' ] || [ "$shaType" = 'sha1' ]
+}
+
+for version in "${versions[@]}"; do
+	export version
+
+	versionAlpineVersion="${alpineVersions[$version]:-$defaultAlpineVersion}"
+	versionDebianSuite="${debianSuites[$version]-$defaultDebianSuite}" # intentionally missing ":" so it can be empty (again, https://github.com/docker-library/postgres/issues/582 😭)
```

Comment:
Another link to `postgres`. I'd guess that we don't need the empty functionality like `postgres` does. I.e., a version of `redis` is unlikely to have indexing issues, so we should be able to move Debian releases without any problem.